ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/rollup/RollupCreatePayloadTest.java

```diff
@@ -0,0 +1,270 @@
+  }
+
+  @Test
+  public void shouldReturnInvalidTerminalBlockIfParentRootHashDoesNotExist() {
+    final var invalidParentHash = Hash.hash(Bytes32.fromHexStringLenient("0x1337deadbeef"));
+
+    assertStatus(
+        invalidParentHash,
+        Collections.emptyList(),
+        RollupCreatePayloadStatus.INVALID_TERMINAL_BLOCK);
+  }
+
+  @Test
+  public void shouldReturnPayloadWhenBlockIsValid() {
+    final BlockHeader mockHeader =
+        new BlockHeaderTestFixture()
+            .prevRandao(mockPrevRandao)
+            .gasLimit(blockGasLimit)
+            .timestamp(mockBlockTimestamp)
+            .coinbase(feeRecipient)
+            .buildHeader();
+    final Block mockEmptyBlock =
+        new Block(mockHeader, new BlockBody(List.of(transaction2), Collections.emptyList()));
+
+    when(blockchain.getBlockHeader(mockHash)).thenReturn(Optional.of(mock(BlockHeader.class)));
+    final var mockTransactionSelectionResults = mock(TransactionSelectionResults.class);
+    when(mockTransactionSelectionResults.getInvalidTransactions())
+        .thenReturn(
+            List.of(
+                new TransactionValidationResult(
+                    transaction1,
+                    ValidationResult.invalid(TransactionInvalidReason.NONCE_TOO_LOW))));
+    var blockCreationResult =
+        new PayloadCreationResult(
+            mockPayloadId,
+            new BlockCreationResult(mockEmptyBlock, mockTransactionSelectionResults),
+            new BlockValidator.Result(new BlockProcessingOutputs(null, Collections.emptyList())));
+    when(mergeCoordinator.createBlock(any(), any(), any(), any(), any()))
+        .thenReturn(blockCreationResult);
+
+    final var result =
+        assertStatus(
+            mockHash,
+            List.of(transaction1, transaction2, transaction3),
+            RollupCreatePayloadStatus.PROCESSED);
+
+    verify(mergeCoordinator)
+        .createBlock(
+            any(),
+            eq(mockBlockTimestamp),
+            eq(feeRecipient),
+            eq(List.of(transaction1, transaction2, transaction3)),
+            eq(mockPrevRandao));
+
+    assertThat(result.getPayloadId()).isNotNull();
+    assertThat(result.getPayloadId()).matches("0[xX][0-9a-fA-F]{16}");
+    assertThat(result.getExecutionPayload()).isNotNull();
+    assertThat(result.getExecutionPayload().getFeeRecipient())
+        .isEqualTo(feeRecipient.toHexString());
+    assertThat(result.getExecutionPayload().getPrevRandao())
+        .isEqualTo(mockEmptyBlock.getHeader().getPrevRandao().get().toHexString());
+    assertThat(result.getInvalidTransactions().size()).isEqualTo(1);
+    assertThat(result.getInvalidTransactions().get(0).getTransaction())
+        .isEqualTo(rlpEncode(transaction1));
+    assertThat(result.getInvalidTransactions().get(0).getInvalidReason())
+        .isEqualTo(TransactionInvalidReason.NONCE_TOO_LOW);
+    assertThat(result.getUnprocessedTransactions()).isEqualTo(List.of(rlpEncode(transaction3)));
```

Comment:
why is transaction3 expected to be invalid? nonce looks legit

Reply:
It's valid yes. In this unit test transaction validation is mocked. We simulate a scenario where the processing result has the 3 possible outcomes for a transaction: 1) Invalid, 2) valid and included in the block, 3) unprocessed because the block was already full (reached the gasLimit). Unprocessed transactions list is for point 3.